saltproc/saltproc.py

```diff
@@ -265,108 +500,166 @@ def write_mat_file(self):
+
+        for group, scheme in self.rep_scheme.items():
+            if scheme['from'] != 'fertile':
+                continue
+            else:
+                qty_to_fill = self.core_mass[scheme['to']
+                                             ] - sum(self.core[scheme['to']])
+                for indx, frac in enumerate(scheme['comp']):
+                    isoid = self.find_iso_indx(scheme['element'][indx])
+                    self.refill(isoid, qty_to_fill*frac, scheme['to'])
+                    print('ADDING IN %f kg of %s to %s' % (
+                        qty_to_fill * frac, scheme['element'][indx], scheme['to']))
+
+    def reactivity_control(self):
+        """ Controls fraction of fissile material
+            input into core to control keff into
+            a range
+        """
+        # check EOC KEFF for determining fissile_add_back_frac
+
+        self.eoc_keff = self.read_res(1)
+        # how much pu we lost:
+        pu = self.find_iso_indx(['Pu'])
+        pu_loss = self.driver_before_db[self.current_step,
+                                        pu] - self.driver_after_db[self.current_step-1, pu]
+        pu_loss = sum(pu_loss)
+        pu_avail = sum(self.fissile_tank_db[self.current_step, :])
+        print('EOC KEFF IS %f +- %f' % (self.eoc_keff[0], self.eoc_keff[1]))
+
+        if self.eoc_keff[0] > 1.05:
+            print('KEFF IS TOO HIGH: NOT PUTTING ANY MORE PU IN DRIVER\n')
+            qty = 0
+        elif self.eoc_keff[0] <= 1.05 and self.eoc_keff[0] > 1.01:
+            print('KEFF IS IN A GOOD SPOT: PUTTING THE AMOUNT LOST FROM PREV DEPLETION\n')
+            qty = min(pu_avail, pu_loss)
+        elif self.eoc_keff[0] <= 1.01:
+            print('KEFF IS LOW: PUTTING IN 1.5 TIMES PU THAN PREIVOUS STEP:')
+            qty = min(self.prev_qty * 1.5, pu_avail)
+        if qty == pu_avail:
+            print('NOT ENOUGH PU AVAILABLE: PUTTING THE MAXIMUM AMOUNT AVAILABLE')
+
+        self.prev_qty = qty
+        return qty
 
     def record_db(self):
         """ Records the processed fuel composition, Keff values,
             waste tank composition to database
         """
-        self.keff_db[:, self.current_step - 1] = self.read_res(1)
-        self.keff_db_0[:, self.current_step - 1] = self.read_res(0)
-        self.bu_adens_db_1[self.current_step, :] = self.core
-        self.tank_adens_db[self.current_step, :] = (self.tank_adens_db[self.current_step - 1, :]
-                                                    + self.rem_adens.sum(axis=0))
-        # store amount of Th tank
-        th232_id = self.find_iso_indx('Th232')
-        prev_th = self.th_adens_db[self.current_step - 1, th232_id]
-        orig_th = self.bu_adens_db_0[0, th232_id]
-        step_th = self.bu_adens_db_0[self.current_step, th232_id]
-        self.th_adens_db[self.current_step,
-                         th232_id] = prev_th - orig_th - step_th
+        self.keff_eoc_db[:, self.current_step - 1] = self.read_res(1)
```

Comment:
these changes seem to deprecate previous saltproc single fluid thorium reactor capabilities.